heareval/tasks/util/luigi.py

```diff
@@ -321,15 +314,22 @@ def run(self):
             assert len(audiofiles) == len(audiodf.drop_duplicates())
 
             # Get the label from the metadata with the help of the slug of the filename
-            sublabeldf = labeldf.merge(audiodf, on="slug")[["slug", "label"]]
-            # Check if all the labels were found from the metadata
-            assert len(sublabeldf) == len(audiofiles)
-            # Save the slug and the label in as the parition metadata
+            sublabeldf = labeldf.merge(audiodf, on="slug")
+
+            if self.data_config["task_type"] == "scene_labeling":
+                # Check if all the labels were found from the metadata
+                assert len(sublabeldf) == len(audiofiles)
+            elif self.data_config["task_type"] == "event_labeling":
+                # This won't work for sound event detection where there might be
+                # zero or more than one event per file
+                pass
+            else:
+                raise ValueError
+
+            # Save the slug and the label in as the split metadata
             sublabeldf.to_csv(
-                os.path.join(self.workdir, f"{partition}.csv"),
-                columns=["slug", "label"],
```

Comment:
if we want to keep these csv files clean we could use slug and label for scene_labeling and slug, label, start, and end for event_labeling -- and then save an additional metadata.csv file that is just the full set of all metadatas for all splits (including the split label).

Reply:
@jorshi Agreed, I think we should have both.